src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueBehavior.cs

```diff
@@ -31,39 +53,36 @@ public MoveFaultsToErrorQueueBehavior(CriticalError criticalError, IPipelineBase
             {
                 throw;
             }
-            catch (Exception exception)
+            catch (Exception e)
             {
-                try
-                {
-                    var message = context.Message;
+                faultsStatusStorage.AddException(uniqueMessageId, e);
```

Comment:
Would it make sense to use [`ExceptionDispatchInfo`](https://msdn.microsoft.com/en-us/library/system.runtime.exceptionservices.exceptiondispatchinfo(v=vs.110).aspx) to not cripple the call stack unnecessarily?

> The ExceptionDispatchInfo object stores the stack trace information and Watson information that the exception contains at the point where it is captured. The exception can be thrown at another time and possibly on another thread by calling the ExceptionDispatchInfo.Throw method. The exception is thrown as if it had flowed from the point where it was captured to the point where the Throw method is called.